### PR TITLE
#17: Proper scaling factor in zoom slider 

### LINKED ISSUE #17: Proper scaling factor in zoom slider

### PR DESCRIPTION
**Pull request by alvinosh**
In scope of
https://github.com/ARK-Builders/ARK-Gallery/issues/17

### CODE REVIEW

File: src/lib/components/Gallery.svelte [hunk 1]
```diff
@@ -5,13 +5,14 @@
...
 </script>
 
 <div
-	class="gap-3 flex-1 w-full overflow-auto
-		{noImage
-		? 'flex flex-col gap-5 items-center w-full justify-center'
-		: 'grid grid-cols-2 md:grid-cols-4 xl:grid-cols-6'}"
+	class="w-full gap-3
+		{noImage ? 'flex flex-col items-center justify-center gap-5' : 'grid grid-flow-row'}"
+	style="grid-template-columns: repeat(auto-fill, minmax({imageHeight}px, 1fr));"
```
Comment (prokawsar) [inline src/lib/components/Gallery.svelte]: @alvinosh 
With this approach, the zooming feature doesn't work smoothly. 
In my test, zoom from 100 to 125, doesn't have any changes to images, which is wrong and bad UX.
Zooming should work as smoothly.

**Review by prokawsar — COMMENTED** [review]
About Index page, seems some css bring to parent component instead of own component, no major changes. 
I think, it's better to maintain clean code on top level component. Attached both code structure as screenshot. What do you think @kirillt ?
Also, we can hide zoom bar for gallery view by a simple login with `galleryStore.galleryView` store.
Screenshot 2024-04-13 at 9 03 03 AM

Screenshot 2024-04-13 at 9 02 19 AM

**Comment by alvinosh** [comment]
> About Index page, seems some css bring to parent component instead of own component, no major changes. I think, it's better to maintain clean code on top level component. Attached both code structure as screenshot. What do you think @kirillt ? Also, we can hide zoom bar for gallery view by a simple login with `galleryStore.galleryView` store. Screenshot 2024-04-13 at 9 03 03 AM
> 
> Screenshot 2024-04-13 at 9 02 19 AM

There is not really a need to maintain "clean-code" at top level component. Later, we want to remove the explicit check 
`class:hidden={!$galleryStore.galleryView}`
and handle it with sveltekit rounding, but we can do that in a sepeate pr
In sveltekit, usually any CSS styles that are related to the page layout, (meaning margins, paddings, flex ect) should be in the layout component. This allows for the components of the page to be more pure and more reusable.

**Comment by kirillt** [comment]
@alvinosh now it feels working to me, thanks.

@prokawsar does it look good enough for you? I have no idea about JS and Svelte code, but if something looks suboptimal but not critical we can merge and raise this question on dev sync.

**Review by prokawsar — APPROVED** [review]
Fixed some minor issues.

**Pull request merged**